### PR TITLE
misc cleanup and changes

### PR DESCRIPTION
**Pull request by Berthalamew**

### CODE REVIEW

File: xlive/H2MOD/Modules/GamePhysics/Patches/ProjectileFix.cpp [hunk 1]
```diff
@@ -155,16 +154,17 @@ void __cdecl matrix4x3_transform_point(void* matrix, real_vector3d* v1, real_vec
 
 void ProjectileFix::ApplyProjectileVelocity()
 {
-	for (auto& proj_tuple : weapon_projectiles)
+	for (uint32 i = 0; i < 3; i++)
```
Comment (nukeulater) [inline xlive/H2MOD/Modules/GamePhysics/Patches/ProjectileFix.cpp]: i < ARRAYSIZE()

File: xlive/H2MOD/Modules/SpecialEvents/SpecialEvents.cpp [hunk 2]
```diff
@@ -8,95 +7,64 @@
...
-		return _special_event_st_paddys;
-
-	if (check_special_event_week(L"12-24") || check_special_event_week(L"12-30") || check_special_event_week(L"1-4"))
-		return _special_event_christmas;
-
-	// One time event
-	/*if (CheckIfEventTime(L"4-12"))
-			return _mook_maddness;*/
 
-	if (check_special_event_week(L"10-20") || check_special_event_week(L"10-27") || check_special_event_date(L"10-31"))
```
Comment (nukeulater) [inline xlive/H2MOD/Modules/SpecialEvents/SpecialEvents.cpp:92]: god bless you

File: xlive/H2MOD/Modules/SpecialEvents/SpecialEvents.cpp [hunk 3]
```diff
@@ -8,95 +7,64 @@
...
-		return _special_event_halloween;
-
-	if (check_special_event_date(L"11-08") || check_special_event_date(L"11-09") || check_special_event_date(L"11-10") || 
-	    check_special_event_date(L"5-30") || check_special_event_date(L"5-31") || check_special_event_date(L"6-01"))
-		return _special_event_birthday;
+	switch (date.month)
+	{
+	// January
+	case 1:
+		event = (IN_RANGE_INCLUSIVE(date.day, 1, 7) ? _special_event_christmas : event);
```
Comment (twist84) [inline xlive/H2MOD/Modules/SpecialEvents/SpecialEvents.cpp:37]: christmas on new years what a concept

Comment (Berthalamew) [reply xlive/H2MOD/Modules/SpecialEvents/SpecialEvents.cpp:37]: Truly is the most wonderful time of the year

**Pull request merged**